homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -975,6 +980,7 @@ async def async_set_percentage(self, percentage: int) -> None:
             )
         self._percentage = percentage
 
+        # NOTE: this is broken, is_on is never set
```

Comment:
Can we fix this too?

Reply:
No need to fix - it's ok with `is_on`. I misunderstood the purpose. is_on is actually a property from one of the classes higher in the hierarchy (in my opionion the code should use composition rather than many levels of inheritance, but this is out of scope for this bugfix)